### PR TITLE
[CMAKE] Avoid double setting of LDFLAGS

### PR DESCRIPTION
**Pull request by Aidyn-A**
It was observed that in some environments `LDFLAGS` gets directly appended to `CMAKE_SHARED_LINKER_FLAGS`. As the result, the same linker flag can appear twice in `CMAKE_SHARED_LINKER_FLAGS` due to manual set:
https://github.com/pytorch/pytorch/blob/1bf4a44b335b53ba9f9c7056a54ed8e4010e561b/CMakeLists.txt#L541-L542
This flag collision causes the build failures at the `cmake` stage.
This PR adds an instruction to `CMakeLists.txt` to avoid double setting of `LDFLAGS` into `CMAKE_SHARED_LINKER_FLAGS`. 

cc @tinglvv @atalman

### CODE REVIEW

**Comment by tinglvv** [comment]
Thanks for cleaning this up. Will this affect ARM CUDA Wheel?

**Comment by Aidyn-A** [comment]
> Thanks for cleaning this up. Will this affect ARM CUDA Wheel?

No, it should not, as I wanted to make it universal

**Comment by tinglvv** [comment]
Sounds good, added the ci/binaries label to test. If the test passes it should be good to go.

**Review by atalman — APPROVED** [review]

**Review by tinglvv — APPROVED** [review]
Looks good to me

**Review by malfet — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -538,8 +538,13 @@ option(BUILD_EXECUTORCH "Master flag to build Executorch" ON)
 if(LINUX)
   set(CMAKE_SHARED_LINKER_FLAGS
       "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-as-needed")
-  set(CMAKE_SHARED_LINKER_FLAGS
-      "${CMAKE_SHARED_LINKER_FLAGS} $ENV{LDFLAGS}")
+
+  set(ENV_LDFLAGS "$ENV{LDFLAGS}")
+  string(STRIP "${ENV_LDFLAGS}" ENV_LDFLAGS)
+  if(NOT ${CMAKE_SHARED_LINKER_FLAGS} MATCHES "${ENV_LDFLAGS}")
```
Comment (malfet) [inline CMakeLists.txt:544]:
```suggestion
  # Do not append linker flags passed via env var if they already there
  if(NOT ${CMAKE_SHARED_LINKER_FLAGS} MATCHES "${ENV_LDFLAGS}")
```

**Pull request closed without merging**